components/sensor/veml7700.rst

```diff
@@ -0,0 +1,203 @@
+
+- If illuminance is higher than 46 lx (100 counts) it gradually reduces integration time to get good reading (best in range 100 - 10,000 counts). 
+- In case of low illuminance (less than 46 lx) it tries to gradually increase gain and only then increase integration time.
+
+Please note, that in low light conditions measurement process might take several seconds due to long exposure periods and sensor reconfigurations.
+
+Lux compensation
+----------------
+
+Lux compensation is done as per recommendation from the manufacturer, however it can be turned off setting ``lux_compensation: false``
+in configuration.
+
+.. math::
+
+    \displaystyle \begin{array}{l}
+    lux & = & counts \times resolution(time, gain)\\
+    lux_{comp} & =& 6.0135e \times 10^{-13} \times lux^4 - 9.3924e \times 10^{-9}  \times  lux^3\\
+    & & + 8.1488e \times 10^{-5}  \times  lux^2 + 1.0023  \times  lux\\
+    \\
+    \text{Where:} & & \\
+    counts & - & \text{sensor readings, counts}\\
+    resolution & - & \text{sensor resolution for given integration time and gain, lx/counts}\\
+    lux & - & \text{calculated illumination, lx}\\
+    lux_{comp} & - & \text{compensated illumniation, lx}\\
+    \end{array}
+
+Available data
+--------------
+
+The implementation offers seven sensors: two providing *lux* value, two - unitless data measurements directly from the device 
+and two - actual *gain* and *integration time* values used for the measurement (useful in automatic mode), and one - fully 
+artificial, somewhat representing near-infrared part of spectre.
+
+- **ambient_light**: Illuminance value for *ALS* channel representing human eye, lx
+- **full_spectrum**: Illuminance value for *WHITE* channel with wide spectrum, lx
+- **infrared**: Calculated illuminance value (*WHITE* minus *ALS*) representing near-infrared spectre, lx
+- **ambient_light_counts**: Raw 16 bit reading from *ALS* channel, counts
+- **full_spectrum_counts**: Raw 16 bit reading from *WHITE* channel, counts
+- **actual_gain**: The actual gain value being used for values reported, multiplier
+- **actual_integration_time**: The actual integration time being used for values reported, ms
+
+Example configuration
+---------------------
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    i2c:
+      # ...
+    sensor:
+      - platform: veml7700
+        address: 0x10
+        update_interval: 60s
+        ambient_light:
+          name: "Ambient light"
+        full_spectrum:
+          name: "Full spectrum"
+        infrared:
+          name: "Calculated near-IR"
+
+
+Configuration variables:
+------------------------
+
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+- **address** (*Optional*, int): Manually specify the I²C address of the device.
+  Defaults to ``0x10``. VEML7700 only uses address ``0x10``. VEML6030 can be either
+  ``0x48`` or ``0x10`` (hardware configuration).
```

Comment:
```suggestion
```
This is explained on the I2C & Sensor pages referenced below (and you already mentioned the addresses used earlier).